cdk/lib/payment-api.ts

```diff
@@ -1,22 +1,409 @@
+          "For the PROD stack you still need to supply this, because PROD instances need access to both PROD and CODE email queues.",
+      }
+    );
+
+    const emailSqsProdArn = new GuStringParameter(
+      this,
+      "EmailSqsQueueProdArn",
+      {
+        description:
+          "For the CODE stack you can leave this empty since it won't be used.  For the PROD stack you need to set it.",
+      }
+    );
+
+    const ophanRole = new GuStringParameter(this, "OphanRole", {
+      description: "ARN of the Ophan cross-account role",
+    });
+
+    const kinesisStreamArn = new GuStringParameter(this, "KinesisStreamArn", {
+      description: "ARN of the kinesis stream to write events to",
+    });
+
+    const contributionsStoreSqsQueueCodeArn = new GuStringParameter(
+      this,
+      "ContributionsStoreSqsQueueCodeArn",
+      {
+        description:
+          "For the PROD stack you still need to supply this, because PROD instances need access to both PROD and CODE contribution store queues.",
+      }
+    );
+
+    const contributionsStoreSqsQueueProdArn = new GuStringParameter(
+      this,
+      "ContributionsStoreSqsQueueProdArn",
+      {
+        description:
+          "For the CODE stack you can leave this empty since it won't be used.  For the PROD stack you need to set it.",
+      }
+    );
+
+    const sqsKmsArn = new GuStringParameter(this, "SqsKmsArn", {
+      description: "ARN of the KMS key for encrypting SQS data",
+    });
+
     const yamlTemplateFilePath = join(
       __dirname,
       "../..",
       "support-payment-api/src/main/resources/cloud-formation.yaml"
     );
+
     new CfnInclude(this, "YamlTemplate", {
       templateFile: yamlTemplateFilePath,
       parameters: {
         Stage: props.stage,
+        EmailSqsQueueCodeArn: emailSqsCodeArn,
+        EmailSqsQueueProdArn: emailSqsProdArn,
+        OphanRole: ophanRole,
+        KinesisStreamArn: kinesisStreamArn,
+        ContributionsStoreSqsQueueCodeArn: contributionsStoreSqsQueueCodeArn,
+        ContributionsStoreSqsQueueProdArn: contributionsStoreSqsQueueProdArn,
+        SqsKmsArn: sqsKmsArn,
+      },
+    });
+
+    // TODO: Should these remain as cloudformation parameters?
+    const projectName = "payment-api";
+    const projectVersion = "0.1";
+
+    const app = "payment-api";
+    const userData = `#!/bin/bash -ev
+    mkdir /etc/gu
+
+    cat > /etc/gu/stage <<'EOF'
+    ${this.stage}
+    EOF
+
+    aws --region ${this.region} s3 cp s3://membership-dist/${this.stack}/${this.stage}/${app}/${projectName}_${projectVersion}_all.deb /tmp
+    dpkg -i /tmp/${projectName}_${projectVersion}_all.deb
+    /opt/cloudwatch-logs/configure-logs application ${this.stack} ${this.stage} ${app} /var/log/${app}/application.log`;
+
+    const playApp = new GuPlayApp(this, {
+      app: "payment-api",
+      access: { scope: AccessScope.PUBLIC },
+      certificateProps: {
+        domainName: props.domainName,
+        hostedZoneId: "Z1E4V12LQGXFEC",
```

Comment:
We should check with DevEx/somebody that understands how certificates work if this is correct. I can see the current template has a [certificate arn](https://github.com/guardian/support-frontend/blob/main/support-payment-api/src/main/resources/cloud-formation.yaml#L91) as a param